components/needs_team/needs_team.jsx

```diff
@@ -81,22 +81,30 @@ export default class NeedsTeam extends React.Component {
         this.state = {
             team,
             finishedFetchingChannels: false,
+            prevProps: this.props
         };
 
         if (!team) {
             this.joinTeam(this.props);
         }
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        if (this.props.match.params.team !== nextProps.match.params.team) {
-            const team = this.updateCurrentTeam(nextProps);
-            this.setState({
-                team,
-            });
-            if (!team) {
-                this.joinTeam(nextProps);
-            }
+    static getDerivedStateFromProps(nextProps, state) {
+        if (state.prevProps.match.params.team !== nextProps.match.params.team) {
+            const team = state.prevProps.teamsList ?
+                state.prevProps.teamsList.find((teamObj) =>
+                    teamObj.name === nextProps.match.params.team) : null;
+            return {prevProps: nextProps, team: (team || null)};
+        }
+        return {prevProps: nextProps};
+    }
+
+    static getSnapshotBeforeUpdate() {
```

Comment:
These should be done in `componentDidUpdate`. You can also compare against the previous state to avoid extra calls to those methods as well